### PR TITLE
让你的单元测试支持testinitialize和cleanup

### PR DESCRIPTION
**Pull request by xinyuehtx**
生活更加美好

### CODE REVIEW

File: src/MSTest.Extensions/Utils/ReflectionHelper.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using System.Reflection;
+
+namespace MSTest.Extensions.Utils
+{
+    /// <summary>
+    /// 反射辅助类
+    /// </summary>
+    public static class ReflectionHelper
```
Comment (walterlv) [inline src/MSTest.Extensions/Utils/ReflectionHelper.cs]: We have `type.GetRuntimeProperty`/`Field`/`Method` to replace all methods in this class.

File: src/MSTest.Extensions/Contracts/ContractTestCaseAttribute.cs [hunk 2]
```diff
@@ -30,17 +31,103 @@ namespace MSTest.Extensions.Contracts
...
+            ReflectionHelper.SetProperty(classInfo, MsTestMemberName.TestClassInfoPropertyBaseTestCleanupMethodsQueue, baseTestCleanupMethodsQueue);
+            ReflectionHelper.SetField(classInfo, MsTestMemberName.TestClassInfoFieldTestCleanupMethod, testCleanupMethod);
+            ReflectionHelper.SetProperty(classInfo, MsTestMemberName.TestClassInfoPropertyBaseTestInitializeMethodsQueue, new Queue<MethodInfo>());
+            ReflectionHelper.SetField(classInfo, MsTestMemberName.TestClassInfoFieldTestInitializeMethod, null);
+            testMethod.Invoke(null);
+            ReflectionHelper.SetProperty(classInfo, MsTestMemberName.TestClassInfoPropertyBaseTestInitializeMethodsQueue, baseTestInitializeMethodsQueue);
+            ReflectionHelper.SetField(classInfo, MsTestMemberName.TestClassInfoFieldTestInitializeMethod, testInitializeMethod);
+        }
+
+        private object GetTestCleanupMethod(object classInfo)
```
Comment (walterlv) [inline src/MSTest.Extensions/Contracts/ContractTestCaseAttribute.cs]: `GetTestCleanupMethod` `GetTestInitializeMethod` `GetBaseTestCleanupMethodsQueue` `GetClassInfo` `GetBaseTestInitializeMethodsQueue`, all these five methods are doing an another thing that should not be done under `ContractTestCaseAttribute`. Please extract these methods and their callers out of `ContractTestCaseAttribute`.

File: src/MSTest.Extensions/Utils/Constants.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿namespace MSTest.Extensions.Utils
+{
+    public static class MsTestMemberName
```
Comment (walterlv) [inline src/MSTest.Extensions/Utils/Constants.cs]: 'M' and 'S' should both be upper case.

File: src/MSTest.Extensions/Contracts/ContractTestCaseAttribute.cs [hunk 4]
```diff
@@ -30,14 +31,19 @@ namespace MSTest.Extensions.Contracts
     [PublicAPI]
     public sealed class ContractTestCaseAttribute : TestMethodAttribute, ITestDataSource
     {
+
```
Comment (walterlv) [inline src/MSTest.Extensions/Contracts/ContractTestCaseAttribute.cs]: Remove this empty line.

File: src/MSTest.Extensions/Contracts/ContractTestCaseAttribute.cs [hunk 5]
```diff
@@ -186,5 +192,11 @@ private void VerifyContracts([NotNull] IList<ITestCase> cases)
         /// because they are not in the same instance.
         /// </summary>
         private int _testCaseIndex;
+        /// <summary>
```
Comment (walterlv) [inline src/MSTest.Extensions/Contracts/ContractTestCaseAttribute.cs]: Add an empty line before this field.

File: src/MSTest.Extensions/Contracts/ContractTestCaseAttribute.cs (hunk 4 above)
Comment (xinyuehtx) [inline src/MSTest.Extensions/Contracts/ContractTestCaseAttribute.cs]: 空行

**Review by walterlv — CHANGES_REQUESTED** [review]

**Pull request merged**